yarn-project/stdlib/src/snapshots/upload.ts

```diff
@@ -0,0 +1,43 @@
+import { pick } from '@aztec/foundation/collection';
+import { jsonStringify } from '@aztec/foundation/json-rpc';
+import type { FileStore } from '@aztec/stdlib/file-store';
+
+import { getBasePath, getSnapshotIndex, getSnapshotIndexPath } from './download.js';
+import type { SnapshotMetadata, SnapshotsIndex } from './types.js';
+
+export async function uploadSnapshot(
+  archiverPath: string,
+  worldStatePath: string,
+  metadata: Pick<SnapshotMetadata, 'l2BlockNumber' | 'l2BlockHash' | 'l1BlockNumber'> &
+    Pick<SnapshotsIndex, 'l1ChainId' | 'l2Version' | 'rollupAddress'>,
+  store: FileStore,
+): Promise<SnapshotMetadata> {
+  const timestamp = Date.now();
+  const date = new Date().toISOString().replace(/[-:T]/g, '').replace(/\..+$/, '');
+
+  const basePath = getBasePath(metadata);
+  const [archiverDataUrl, worldStateDataUrl] = await Promise.all([
+    store.upload(`${basePath}/archiver-${date}-${metadata.l2BlockHash}.db`, archiverPath, { compress: true }),
```

Comment:
Is there a strong deason to have the date in the name, at a first glance it would make downloading more difficult

Reply:
Ah i see there is one index.json that points to the most recent snapshot